lib/stdlib/src/gen_server.erl

```diff
@@ -2568,10 +2580,13 @@ handle_timeout(ServerData, T, M, HibInf) ->
             error
     end.
 
-handle_timeout(ServerData, T, M, HibInf, [], Abs) when ?is_timeout(Abs, T) ->
+handle_timeout(ServerData, T, M, HibInf, {abs, Abs}, _Abs) when is_boolean(Abs),
+								?is_timeout(Abs, T) ->
```

Comment:
This opens up for an improper list with `{abs, Abs}` in the tail.

To not accept improper lists I tend to use this pattern:
``` erlang
handle_action(ServerData, Action) ->
    :
        {timeout, T, M, Opts} ->
            handle_timeout(ServerData, T, M, infinity, listify(Opts), false);
    :
    end.

-compile({inline, [listify/1]}).
listify(Item) when is_list(Item) ->  Item;
listify(Item)                    -> [Item].
```